controllers/profiles/preview/states_preview.go

```diff
@@ -210,7 +211,15 @@ func (h *deployWithBuildWorkflowState) Do(ctx context.Context, workflow *operato
 	}
 
 	// didn't change, business as usual
-	return NewDeploymentReconciler(h.StateSupport, h.ensurers).reconcileWithImage(ctx, workflow, build.Status.ImageTag)
+	result, objs, err := NewDeploymentReconciler(h.StateSupport, h.ensurers).reconcileWithImage(ctx, workflow, build.Status.ImageTag)
+	if err != nil {
+		workflow.Status.Manager().MarkFalse(api.DeployedConditionType, api.DeploymentFailureReason, fmt.Sprintf("Error in deploy the workflow:%s", err))
+		_, err = h.PerformStatusUpdate(ctx, workflow)
+		return result, nil, err
+	}
+	workflow.Status.Manager().MarkTrue(api.DeployedConditionType)
```

Comment:
So this is the only place we set this condition to `true`. Can't we keep using `Running`? Otherwise, if we were to introduce this new status, we need to make sure that we set it accordingly in every state.

| State | Status | Description
|------|--------|------------|
|Running | TRUE | The workflow is running 
|Deployed | TRUE | The workflow has been deployed
|Built | FALSE | The workflow has not being built

So, we need to justify and think about this matrix. If it's not deployed, hence not running, right? That's what I meant by introducing additional complexity to the algorithm by adding a new condition.

Reply:
Sounds good to me. Updated.